### DISCUSSION TITLE
Extending will cause infinite loop if adding a wrapper node (3.13.4)

### DISCUSSION THREAD

**Discussion by gundam78** [alpinejs/alpine#3994] [7. Bugs (archived)] [answered]
This bug only appears in new version 3.13.4 ( 3.13.3 will not)

I write a custom directive to wrapping a node with a <div> ( function to add tooltip or something). It will cause infinite loop, maybe because the original node (with the directive) moving trigger the directive to evaluate again and again?

This demo code using tailwind & daisyui (or see: https://jsfiddle.net/gundam78/be9hrs0L/62/)

```javascript
document.addEventListener('alpine:init', () => {
  Alpine.directive('test', el => {
    console.log('testing');
    let wrapper = document.createElement('div');
    wrapper.setAttribute('class', 'bg-blue-300 p-8');
    el.parentNode.replaceChild(wrapper, el);
    wrapper.appendChild(el);
  });
});
```

```html
<head>
  <link href="https://cdn.jsdelivr.net/npm/daisyui@4.6.0/dist/full.min.css" rel="stylesheet" type="text/css" />
  <script src="https://cdn.tailwindcss.com"></script>
  <script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.13.4/dist/cdn.min.js"></script>
</head>


<div x-data>
  <div class="bg-red-300 w-32 h-32" x-test>
  </div>
</div>
```

**Comment by ekwoka** [comment, marked as answer]
the changes to the mutation observer code seem to be causing this.

**Reply by calebporzio** [reply]
Just tagged 3.13.5 with @ekwoka's fix - hopefully that fixes it

**Reply by gundam78** [reply]
3.13.5 fixed

**Reply by Matsa59** [reply]
Yes I confirm I introduce this bug sorry :/

but the fix 3.13.5 reintroduce the bug on the mutation :/
